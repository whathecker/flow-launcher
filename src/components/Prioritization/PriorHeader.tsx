/* eslint-disable @typescript-eslint/no-unused-vars */
import React from "react";
import { StyleSheet, Image } from "react-native";
import { View, Text } from "../Themed";
import { CloseIcon } from "../shared";
import { Container, Typography } from "../../styles";
import { navigationRef, shortenText } from "../../utils";

type PriorHeaderProps = {
  title: string;
};

const PriorHeader: React.FC<PriorHeaderProps> = ({
  title,
}: PriorHeaderProps) => {
  return (
    <View style={styles.wrapper}>
      <View style={styles.closeIconAreaWrapper}>
        <View style={styles.closeIconWrapper}>
          <CloseIcon pressHandler={() => navigationRef.goBack()} />
        </View>
      </View>
      <View style={styles.headerAreaWrapper}>
        <View style={styles.headerWrapper}>
          <Text style={styles.headerText}>{`Prioritizing`}</Text>
          <Image
            style={styles.image}
            source={require(`../../../assets/images/bar-chart_1f4ca.png`)}
          />
        </View>
      </View>
      <View style={styles.goalAreaWrapper}>
        <Text style={styles.goalText}>{shortenText(title, 50)}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    marginTop: "6%",
    marginBottom: "10%",
  },
  closeIconAreaWrapper: {
    ...Container.flexStart,
    height: "8%",
  },
  closeIconWrapper: {
    paddingLeft: "5%",
    paddingTop: "12%",
    paddingBottom: "5%",
  },
  headerAreaWrapper: {
    marginTop: "10%",
    paddingLeft: "10%",
  },
  headerWrapper: {
    ...Container.flexStart,
    width: "100%",
  },
  headerText: {
    ...Typography.h4,
    fontSize: 20,
    paddingRight: "3%",
  },
  image: {
    width: 32,
    height: 32,
  },
  goalAreaWrapper: {
    width: "90%",
    paddingLeft: "10%",
    marginTop: "8%",
  },
  goalText: {
    ...Typography.h4,
    fontSize: 16,
    paddingBottom: "5%",
  },
  motivationText: {
    ...Typography.p,
    fontSize: 14,
  },
});

export default PriorHeader;
