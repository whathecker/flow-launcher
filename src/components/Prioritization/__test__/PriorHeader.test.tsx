import * as React from "react";
import renderer from "react-test-renderer";
import PriorHeader from "../PriorHeader";

describe("Test PriorHeader component", () => {
  it("Should renders PriorHeader correctly", () => {
    const tree = renderer
      .create(<PriorHeader title="Mock goal" motivation="Mock motivation" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
