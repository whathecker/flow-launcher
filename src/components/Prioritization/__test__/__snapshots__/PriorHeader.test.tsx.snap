// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PriorHeader component Should renders PriorHeader correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "marginBottom": "10%",
        "marginTop": "6%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "8%",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "paddingBottom": "5%",
            "paddingLeft": "5%",
            "paddingTop": "12%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 18,
              "width": 18,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "marginTop": "10%",
          "paddingLeft": "10%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "fontFamily": "Lexend-SemiBold",
              "fontSize": 20,
              "paddingRight": "3%",
            },
          ]
        }
      >
        Prioritizing
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 32,
            "width": 32,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "marginTop": "8%",
          "paddingLeft": "10%",
          "width": "90%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "fontFamily": "Lexend-SemiBold",
            "fontSize": 16,
            "paddingBottom": "5%",
          },
        ]
      }
    >
      Mock goal
    </Text>
  </View>
</View>
`;
